### DISCUSSION TITLE
APNG to GIF

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7654] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

How do I convert a APNG file to a GIF one? What compression settings can I use for the process to lower file size?

**Comment by fmw42** [comment]
If animated, then 

`magick.exe APNG:image.png -coalesce -layers optimize image.gif`

**Reply by Jay123210599** [reply]
What does coalesce do?

**Reply by fmw42** [reply]
It decompresses. In gif files the frames are not full size as constant regions are removed to save space, but the sizes are known so can be expanded again. (simply explanation - actually more complex).  Look at the sizes of your frames and see if they are not all the same.  I am not quite sure about APNG, but GIFs are like that. I put -coalesce in to be sure if APNGs are like gifs. If you find all the APNG frames are all the same size, then you can remove the -coalesce. Read the documentation on an options page. See https://imagemagick.org/script/command-line-options.php#coalesce.  Do you not know about https://imagemagick.org/script/command-line-options.php
